src/Nethermind/Nethermind.Core/ExecutionRequest/ExecutionRequest.cs

```diff
@@ -0,0 +1,95 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+
+using System;
+using System.Collections.Generic;
+using System.Data.SqlTypes;
+using System.Security.Cryptography;
+using System.Linq;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+
+namespace Nethermind.Core.ExecutionRequest;
+
+public enum ExecutionRequestType : byte
+{
+    Deposit = 0,
+    WithdrawalRequest = 1,
+    ConsolidationRequest = 2
+}
+
+public class ExecutionRequest
+{
+    public byte RequestType { get; set; }
+    public byte[]? RequestData { get; set; }
+
+    public byte[] FlatEncode()
+    {
+        byte[] encoded = new byte[RequestData!.Length + 1];
+        encoded[0] = RequestType;
+        RequestData.CopyTo(encoded, 1);
+        return encoded;
+    }
```

Comment:
This is very allocatey, we could change it in multiple ways (not all are compatible with each other0:
a) Keep request in flat-encoded form, RequestType would then be first byte and Data (as Memory or Span) the rest.
b) Provide a buffer which this should just copy to, rather than allocating new array.
c) Get RequestData as Memory<byte> from bigger memory chunk